protostar/cheatable_starknet/controllers/contracts.py

```diff
@@ -77,6 +85,26 @@ class DeployedContract:
     contract_address: int
 
 
+class NonValidatedInternalDeclare(InternalDeclare):
+    # pylint: disable=too-many-ancestors
+
+    def to_external(self) -> DeprecatedDeclare:
+        # It is not implemented in the InternalDeclare itself as wll
+        raise NotImplementedError(
+            "Cannot convert internal declare transaction to external object."
+        )
+
+    def run_validate_entrypoint(
+        self,
+        remaining_gas: int,
+        state: SyncState,
+        resources_manager: ExecutionResourcesManager,
+        general_config: StarknetGeneralConfig,
+    ) -> Tuple[Optional[CallInfo], int]:
+        # This hack might interfere with other things
```

Comment:
If we're doing a hack, can we explain what it hacks? It might help with debugging some day